doc/build-helpers/fixed-point-arguments.chapter.md

```diff
@@ -0,0 +1,184 @@
+
+As a wrapper around `<pkg>.overrideAttrs`, `lib.extendMkDerivation` passes the whole set of arguments directly to the base build helper before extending them, making it incompatible to build helpers with no-passing arguments.
+
+To workaround this issue, [`lib.customisation.adaptMkDerivation`](#function-library-lib.customisation.adaptMkDerivation) is introduced. Instead of taking an attribute overlay that returns a subset of attributes to update, it takes an argument set adapter that returns the whole set of arguments to pass to the base build helper, allowing the removal of some arguments. This also minimize the expression change needed to adopt `lib.adaptMkDerivation` is also smaller, enabling a smooth transition toward fixed-point arguments support.
+
+We could gradually move those no-passing arguments to `passthru`, deprecate custom overriders, and migrate to `lib.extendMkDerivation` later on.
+
+The following is an example of conventional build helper definition with unable-to-pass arguments.
+
+:::{.example #ex-build-helpers-mkLocalDerivation-adaptMkDerivation}
+
+# Example definition of `mkLocalDerivation` extended from `stdenv.mkDerivation` with `lib.adaptMkDerivation`
+
+Should the original definition of build helper `mkLocalDerivation` takes an argument `specialArg` that cannot be passed to the build helper,
+
+```nix
+{ preferLocalBuild ? true
+, allowSubstitute ? false
+, specialArg ? (_: false)
+, ...
+}@args:
+
+stdenv.mkDerivation ({
+  inherit
+    preferLocalBuild
+    allowSubstitute
+    ;
+  # Some expressions involving specialArg
+  greeting = if specialArg "hi" then "hi" else "hello";
+} // removeAttrs [
+  "specialArg"
+] args)
+```
+:::
+
+wrap around the original definition with `lib.adaptMkDerivation` to make the result build helper also accept fixed-point arguments.
+
+```nix
+lib.adaptMkDerivation stdenv.mkDerivation (finalAttrs:
+
+{ preferLocalBuild ? true
+, allowSubstitute ? false
+, specialArg ? (_: false)
+, ...
+}@args:
+
+# The arguments to pass
+{
+  inherit
+    preferLocalBuild
+    allowSubstitute
+    ;
+  # Some expressions involving specialArg
+  greeting =
+    if specialArg "hi" then "hi" else "hello";
+} // removeAttrs [
+  "specialArg"
+] args)
+```
+
+We could then refactor the definition to pass everything properly, while keeping some backward compatibility.
+
+```nix
+lib.adaptMkDerivation stdenv.mkDerivation (finalAttrs:
+
+{ preferLocalBuild ? true
+, allowSubstitute ? false
+, specialArg ? (_: false)
+, ...
+}@args:
+
+# The arguments to pass
+{
+  inherit
+    preferLocalBuild
+    allowSubstitute
+    ;
+  passthru = {
+    specialArg =
+      if (args?specialArg) then
+        # For backward compatibility only.
+        # TODO: Convert to throw after XX.XX branch-off.
+        lib.warn
+          "mkLocalDerivation: Expect specialArg under passthru."
+          args.specialArg
+      else
+        (_: false);
+  } // passthru;
```

Comment:
```suggestion
  } // args.passthru or { };
```

Reply:
Oops! Thank you for noticing the glitch!